packages/flutter/lib/src/painting/inline_span.dart

```diff
@@ -407,3 +434,287 @@ abstract class InlineSpan extends DiagnosticableTree {
+  final ui.TextBaseline? textBaseline;
+
+  /// An attribute which overwrites an [InlineSpan]'s [TextStyle.wordSpacing]
+  /// if set to non-null.
+  final double? wordSpacing;
+
+  /// An attribute which overwrites an [InlineSpan]'s [TextStyle.letterSpacing]
+  /// if set to non-null.
+  final double? letterSpacing;
+
+  /// An attribute which overwrites an [InlineSpan]'s [TextStyle.color] if set to
+  /// non-null.
+  ///
+  /// Must be null if [foreground] is set to a non-null value.
+  final ui.Color? color;
+
+  /// An attribute which overwrites an [InlineSpan]'s [TextStyle.foreground] if
+  /// set to non-null.
+  ///
+  /// Must be null if [color] is set to a non-null value.
+  final ui.Paint? foreground;
+
+  /// An attribute which overwrites an [InlineSpan]'s [TextStyle.backgroundColor]
+  /// if set to non-null.
+  ///
+  /// Must be null if [background] is set to a non-null value.
+  final ui.Color? backgroundColor;
+
+  /// An attribute which overwrites an [InlineSpan]'s [TextStyle.background] if
+  /// set to non-null.
+  ///
+  /// Must be null if [backgroundColor] is set to a non-null value.
+  final ui.Paint? background;
+
+  /// An attribute which overwrites an [InlineSpan]'s [TextStyle.shadows] if set
+  /// to non-null.
+  final List<ui.Shadow>? shadows;
+
+  /// An attribute which overwrites the [TextDecoration.underline] aspect of
+  /// [TextStyle.decoration] on an [InlineSpan], if set to non-null.
+  ///
+  /// Setting this value to true applies underline and setting it to false
+  /// disables underline.
+  final bool? underline;
+
+  /// An attribute which overwrites the [TextDecoration.overline] aspect of
+  /// [TextStyle.decoration] on an [InlineSpan], if set to non-null.
+  ///
+  /// Setting this value to true applies overline and setting it to false
+  /// disables overline.
+  final bool? overline;
+
+  /// An attribute which overwrites the [TextDecoration.lineThrough] aspect of
+  /// [TextStyle.decoration] on an [InlineSpan], if set to non-null.
+  ///
+  /// Setting this value to true applies lineThrough and setting it to false
+  /// disables lineThrough.
+  final bool? lineThrough;
+
+  /// An attribute which overwrites an [InlineSpan]'s [TextStyle.decorationColor]
+  /// if set to non-null.
+  final ui.Color? decorationColor;
+  /// An attribute which overwrites an [InlineSpan]'s [TextStyle.decorationStyle]
+  /// if set to non-null.
+  final ui.TextDecorationStyle? decorationStyle;
+  /// An attribute which overwrites an [InlineSpan]'s [TextStyle.decorationThickness]
+  /// if set to non-null.
+  final double? decorationThickness;
+
+  /// An attribute which overwrites [TextSpan.recognizer] if set to non-null.
+  ///
+  /// When a recognizer is specified, the [InlineSpan.updateAttributesAtOffset] method
+  /// replaces existing recognizers (if any) within the given range with the
+  /// specified recognizer.
+  ///
+  /// When this value is set to [remove], the [InlineSpan.updateAttributesAtOffset]
```

Comment:
Wondering: Could we use null to indicate "remove" by using `remove` (under another name) as the default value for this property? That default value could maybe even have a private type so we don't need to export RemoveInlineSpanAttribute.

Reply:
Since the sentinel values aren't exposed, whether an attribute should be updated would become a compile-time decision? 
E.g.:
```dart
final Color? maybeColor = ...;
if (maybeColor == null) {
  updateAttributes(.... other attributes);
} else { 
  updateAttributes(.... other attributes, color: maybeColor);
}
```
It will quickly become unscaleable configuration matrix I think,  not great for APIs that want to wrap this API.